src/layouts/index.jsx

```diff
@@ -66,6 +68,29 @@ class Layout extends React.Component {
     );
   }
 
+  getQuestionNumbersByPage() {
+    const { pages } = this.state;
+    const qsByPage = {};
+    let qNum = 0;
+
+    // eslint-disable-next-line consistent-return
+    pages.forEach(page => {
+      const { questionsByPage: questions, id } = page;
+
+      if (questions) {
+        qsByPage[id] = questions.map(() => {
+          qNum += 1;
+
+          return qNum;
+        });
+      } else {
+        qsByPage[id] = [];
+      }
+    });
+
+    return qsByPage;
+  }
+
```

Comment:
So here I updated your getPageNumbers method to do basically the same thing, just as a loop within the loop so we could get the question numbers for every page and then spit out an object where the page ids are the keys and the values are arrays of question numbers (or empty arrays).